lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3904,9 +3904,39 @@ namespace {
     }
 
     Decl *VisitUsingShadowDecl(const clang::UsingShadowDecl *decl) {
-      // Using shadow declarations are not imported; rather, name lookup just
-      // looks through them.
-      return nullptr;
+      Optional<ImportedName> correctSwiftName;
```

Comment:
Can you add a check right at the beginning to only handle UsingShadowDecls that refer to types? We probably should handle functions et al someday as well, but per your focus here the types are most important.